src/overlays/actors/ovl_En_Mnk/z_en_mnk.c

```diff
@@ -28,191 +55,2186 @@ ActorInit En_Mnk_InitVars = {
+    &object_mnk_Anim_003584, &object_mnk_Anim_005194, &object_mnk_Anim_0052C4,
+    &object_mnk_Anim_003854, &object_mnk_Anim_005390, &object_mnk_Anim_005414,
+    &object_mnk_Anim_0037E8, &object_mnk_Anim_0054B4, &object_mnk_Anim_003504,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mnk/func_80AB9708.s")
+AnimationHeader* sHangingMonkeyAnimations[] = {
+    &object_mnk_Anim_0062D8,
+    &object_mnk_Anim_007380,
+    &gHangingMonkeyStruggleAnim,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mnk/func_80AB973C.s")
+AnimationHeader* sHangingMonkeyRopeAnimations[] = {
+    &object_mnk_Anim_01C1B8,
+    &object_mnk_Anim_01C1F8,
+    &gHangingMonkeyRopeStruggleAnim,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mnk/func_80AB977C.s")
+AnimationHeader* sMonkeyAnimations[] = {
+    &object_mnk_Anim_0105DC, &object_mnk_Anim_009FE0, &object_mnk_Anim_0099B0, &object_mnk_Anim_0095B4,
+    &object_mnk_Anim_0095B4, &object_mnk_Anim_00B864, &object_mnk_Anim_00B864, &object_mnk_Anim_009CC0,
+    &object_mnk_Anim_009CC0, &object_mnk_Anim_008814, &object_mnk_Anim_005A08, &object_mnk_Anim_01A4F8,
+    &object_mnk_Anim_01B468, &object_mnk_Anim_01BB0C, &object_mnk_Anim_01C17C,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mnk/func_80AB97B4.s")
+u32 D_80AB9DD8[] = { 2, 2, 3, 4, 5, 6, 100 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mnk/func_80AB99D4.s")
+Color_RGBA8 sHangingMonkeySplashPrimColor = { 255, 255, 155, 255 };
+Color_RGBA8 sHangingMonkeySplashEnvColor = { 255, 100, 0, 255 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mnk/EnMnk_Draw.s")
+static Vec3f sMonkeyFocusPosOffset = { 300.0f, 400.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mnk/func_80AB9BAC.s")
+TexturePtr sMonkeyFaceTextures[] = {
+    object_mnk_Tex_015020, object_mnk_Tex_015920, object_mnk_Tex_016120, object_mnk_Tex_016920,
+    object_mnk_Tex_017120, object_mnk_Tex_017920, object_mnk_Tex_018120,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mnk/func_80AB9C4C.s")
+void EnMnk_TiedUpMonkey_ChangeAnim(EnMnk* this, s32 animIndex, u8 mode, f32 transitionRate) {
+    Animation_Change(&this->skelAnime, sTiedUpMonkeyAnimations[animIndex], 1.0f, 0.0f,
+                     Animation_GetLastFrame(&sTiedUpMonkeyAnimations[animIndex]->common), mode, transitionRate);
+    Animation_Change(&this->propSkelAnime, sTiedUpMonkeyPoleAnimations[animIndex], 1.0f, 0.0f,
+                     Animation_GetLastFrame(&sTiedUpMonkeyPoleAnimations[animIndex]->common), mode, transitionRate);
+
+    switch (animIndex) {
+        case 0:
+            this->unk_3E0 = 5;
+            break;
+
+        case 6:
+            this->unk_3E0 = 4;
+            break;
+
+        case 3:
+        case 8:
+            this->unk_3E0 = 6;
+            break;
+
+        default:
+            this->unk_3E0 = 0;
+            break;
+    }
+    this->animIndex = animIndex;
```

Comment:
You have the anim enum, could be used here (or in another documentation pass if that was going to happen)